src/Var.cc

```diff
@@ -405,7 +406,13 @@ static void make_var(const IDPtr& id, TypePtr t, InitClass c, ExprPtr init,
 void add_global(const IDPtr& id, TypePtr t, InitClass c, ExprPtr init,
                 std::unique_ptr<std::vector<AttrPtr>> attr, DeclType dt)
 	{
-	make_var(id, std::move(t), c, std::move(init), std::move(attr), dt, true);
+	bool do_init = activation_mgr->IsActivated();
+
+	if ( dt == VAR_REDEF && ! activation_mgr->AddingRedef(id, c, init, attr) )
+		// Don't actually change the attributes.
+		attr = nullptr;
```

Comment:
(FYI, I don't think that snippet is quite correct, because I think it's possible for there to be attributes being set even though `do_init` is false.)